### PR TITLE
Automation Testing via Nightwatchjs

### PR DESCRIPTION
**Pull request by ankitguptaindia**
Automation Testing via Nightwatchjs Guide

### CODE REVIEW

File: content/testing/functional.md [hunk 1]
```diff
@@ -1,37 +1,31 @@
...
-2. Do we need cross-browser testing in nodejs?
-3. Cookie/session support
-4. File upload support
-5. AJAX support
+1.	[phantomjs](http://phantomjs.org/) - just a browser
+2.	[theintern.io](http://theintern.io/#compare) - has too many features and still not much talked about. Need to check why?
+3.	[karma](http://karma-runner.github.io/) - from Angular JS team. They have one more
+4.	[nightwatchjs](http://nightwatchjs.org/) - another famius tool
+
+**Evaluation criteria** 1. Do we need unit tetsing in nodejs? 2. Do we need cross-browser testing in nodejs? 3. Cookie/session support 4. File upload support 5. AJAX support
```
Comment (rahul286) [inline content/testing/functional.md]: Looks like your PR removed line breaks

File: content/testing/functional.md [hunk 2]
```diff
@@ -1,37 +1,31 @@
...
-3. [karma](http://karma-runner.github.io/) - from Angular JS team. They have one more
-4. [nightwatchjs](http://nightwatchjs.org/) - another famius tool
-
-**Evaluation criteria**
-1. Do we need unit tetsing in nodejs?
-2. Do we need cross-browser testing in nodejs?
-3. Cookie/session support
-4. File upload support
-5. AJAX support
+1.	[phantomjs](http://phantomjs.org/) - just a browser
```
Comment (rahul286) [inline content/testing/functional.md]: Why spaces are added here?

Please do not make unnecessary changes to line breaks and spaces

File: content/testing/functional.md [hunk 3]
```diff
@@ -1,37 +1,31 @@
...
+2.	Node.js based
 
 ### Compare
 
-| Selenium | node.js |
-| -- | -- |
-| Tests are written in JAVA | Tests are written in nodejs |
-| Complicated to setup | Much easier to setup |
-| Tools are finalized | Exacts tools are yet to finalize |
+| Selenium | node.js | | -- | -- | | Tests are written in JAVA | Tests are written in nodejs | | Complicated to setup | Much easier to setup | | Tools are finalized | Exacts tools are yet to finalize |
```
Comment (rahul286) [inline content/testing/functional.md]: line break

File: content/testing/functional.md [hunk 4]
```diff
@@ -1,37 +1,31 @@
...
+1.	cross-browser (css/js front-end perfection)
+2.	single-browser (raw functioanlity tests)
 
-## Tools
+Tools
+-----
 
-1. Selenium
-2. Node.js based
+1.	Selenium
```
Comment (rahul286) [inline content/testing/functional.md]: spaces

File: content/testing/functional.md [hunk 5]
```diff
@@ -1,37 +1,31 @@
...
-2. single-browser (raw functioanlity tests)
+Two main categories:
+--------------------
 
+1.	cross-browser (css/js front-end perfection)
+2.	single-browser (raw functioanlity tests)
 
-## Tools
+Tools
+-----
```
Comment (rahul286) [inline content/testing/functional.md]: Its seems you are using some markdown tool?

Please use gitbook editor or some other editor which doesn't do auto-formatting like this.

File: content/testing/functional.md (hunk 1 above)
Comment (ankitguptaindia) [inline content/testing/functional.md]: I used Atom editor for writing. It automatically added spaces/line break.  
I'll update this.

Comment (rahul286) [inline content/testing/functional.md]: I think this PR doesn't make any changes to `content/testing/functional.md`

Also line 31 like above will break list as markdown need each list item on separate line.

So please correct `content/testing/functional.md` or revert your changes to `content/testing/functional.md`

**Comment by ankitguptaindia** [comment]
@rahul286 

I have revert the auto formatting changes under functional.md.
Please review and merge the changes.

**Comment by rahul286** [comment]
Thanks for your efforts. :-)
